File: python/another.py
import argparse
import sys


def count_lines(reader):
    counter = 0
    for _ in reader:
        counter += 1
    return counter


def count_bytes(reader):
    return len(reader.read())


def count_runes(reader):
    counter = 0
    for line in reader:
        counter += len(line)
    return counter


def count_words(reader):
    counter = 0
    for line in reader:
        counter += len(line.split())
    return counter


def count(reader, c_lines, c_bytes, c_runes):
    if c_lines:
        return count_lines(reader)

    if c_bytes:
        return count_bytes(reader)

    if c_runes:
        return count_runes(reader)

    return count_words(reader)


def get_args():
    parser = argparse.ArgumentParser()
    parser.add_argument(
        "-l", "--lines", action="store_true", help="Count lines")
    parser.add_argument(
        "-b", "--bytes", action="store_true", help="Count bytes")
    parser.add_argument(
        "-r", "--runes", action="store_true", help="Count runes")
    args = parser.parse_args()

    return args


def main():
    args = get_args()
    reader = sys.stdin
    result = count(reader, args.lines, args.bytes, args.runes)
    print(result)


if __name__ == "__main__":
    main()

File: typescript/src/another.ts
import * as readline from 'readline';

import { Command } from 'commander';

class Config {
  constructor(
    public countLines: boolean,
    public countBytes: boolean,
    public countRunes: boolean
  ) {}
}

// Parse the command-line arguments and return a Config object
function getArgs(): Config {
  const program = new Command('wc');
  program
    .option('-l, --lines', 'Count lines')
    .option('-b, --bytes', 'Count bytes')
    .option('-r, --runes', 'Count runes')
    .parse(process.argv);

  const args = program.opts<{
    lines?: boolean;
    bytes?: boolean;
    runes?: boolean;
  }>();

  return new Config(
    args.lines || false,
    args.bytes || false,
    args.runes || false
  );
}

// Count the number of lines in the input
function countLines(input: string): number {
  return input.split('\n').length - 1;
}

// Count the number of bytes in the input
function countBytes(input: string): number {
  return Buffer.byteLength(input);
}

// Count the number of runes in the input
function countRunes(input: string): number {
  return input.length;
}

// Count the number of words in the input
function countWords(input: string): number {
  return input.split(/\s+/).length;
}

// Count the lines, bytes, runes, or words in the input
function count(input: string, config: Config): number {
  if (config.countLines) {
    return countLines(input);
  } else if (config.countBytes) {
    return countBytes(input);
  } else if (config.countRunes) {
    return countRunes(input);
  } else {
    return countWords(input);
  }
}

// Read input from stdin and print the result
function run(config: Config): void {
  const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
    terminal: false,
  });

  let input = '';
  rl.on('line', (line: string) => {
    input += line + '\n';    
  });
  
  rl.on('close', () => {
    console.log(count(input, config));
  });

  rl.on('error', (err: any) => {
    console.log(err);
  });
}

// Parse the command-line arguments and execute the program
const args = getArgs();
run(args);
